picard/match_details_log.py

```diff
@@ -0,0 +1,240 @@
+                    else:
+                        self.rel_dt = dt
+                else:
+                    self.rel_dt = ""
+
+                self.med_tot_cnt = release.get('medium-count', 0)
+                self.rel_tot_trk_cnt = release.get('track-count', 0)
+
+                if 'media' in release:
+                    medium = release['media'][0]
+                    self.med_fmt = medium.get('format', "")
+                    self.med_pos = medium.get('position', 0)
+                    self.med_trk_cnt = medium.get('track-count', 0)
+                    self.trk_pos = medium['tracks'].get('position', '') if 'tracks' in medium else 0
+                    self.trk_id = medium['tracks'].get('id', '') if 'tracks' in medium else ""
+                else:
+                    self.med_fmt = ""
+                    self.med_pos = 0
+                    self.med_trk_cnt = 0
+                    self.trk_pos = 0
+                    self.trk_id = ""
+
+            self.parts = parts
+            self.release_parts = release_parts
+            self.recording_base_score = recording_base_score
+            self.similarity = similarity
+
+            add_match_dtl_entry(self)
+
+        except (KeyError):
+            log.error(traceback.format_exc())
+
+    def __str__(self):
+        return self.fmt_with_labels()
+
+    @staticmethod
+    def get_hdr_tsv():
+        hdr = ("nbr\tmsg\tis_better\tsimilarity\tresult_id\t"
+               "recording_id\trec_title\trec_len\trec_src_cnt\trec_artist\t"
+               "relgrp_id\trelgrp_title\tprim-type\tsec-type\t"
+               "rel_id\tcountry\tdate\t"
+               "media_fmt\tmed_pos\tmed_tot_cnt\t"
+               "track_id\ttrk_pos\tmedia_trk_cnt\trel_tot_trk_cnt\t"
+               "Score components: parts\tScore components: release parts\trecording_base_score")
+        return hdr
+
+    def fmt_as_tsv(self):
+        fmt = ("{m.match_dtl_entry_type:s}\t"
+               "{m.is_new_str:s}\t"
+               "{m.similarity:.4f}\t"
+               "{m.acoust_id:s}\t"
+               "{m.rec_id:s}\t"
+               "{m.rec_title:s}\t"
+               "{m.rec_len_str:s}\t"
+               "{m.rec_src_cnt:3d}\t"
+               "{m.rec_artist:s}\t"
+               "{m.rg_id:s}\t"
+               "{m.rg_title:s}\t"
+               "{m.rg_pri_type:s}\t"
+               "{rg_sec_type:s}\t"
+               "{m.rel_id:s}\t"
+               "{m.rel_ctry:s}\t"
+               "{m.rel_dt:s}\t"
+               "{m.med_fmt:s}\t"
+               "{m.med_pos:2d}\t"
+               "{m.med_tot_cnt:2d}\t"
+               "{m.trk_id:s}\t"
+               "{m.trk_pos:2d}\t"
+               "{m.med_trk_cnt:2d}\t"
+               "{m.rel_tot_trk_cnt:2d}\t"
+               "{parts:s}\t"
+               "{release_parts:s}\t"
+               "{m.recording_base_score:5.3f}")
+
+        fmtd_str = fmt.format(m=self,
+                              rg_sec_type=str(self.rg_sec_type),
```

Comment:
Those explicit conversion to `str` should not be needed.

Reply:
seems with out it I get TypeError: unsupported format string passed to list.__format__  
I tried some other variations and I get KeyError.  Please share if you know how to do it.

Reply:
Likely because `self.rg_sec_type` isn't a string, but a list. But I think the conversion should be done once when setting the property.

https://github.com/metabrainz/picard/pull/1497/files#diff-ce85bb5328861d2a80f6a7abefcf2338R119

Reply:
Done, conversion done once when creating object.  Also took concept further that we should not have to execute format() more than once either.  so doing it once now when object is created.